### ISSUE TITLE
Enhancement: Warn about negation with `set -e`

### ISSUE THREAD

**Issue by kbulygin** [koalaman/shellcheck#1531]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
set -e
! grep something_unexpected some_file
echo ok
```

#### Here's what shellcheck currently says:
```
No issues detected!
```

#### Here's what I wanted or expected to see:
```
`!` is probably unintentional.
```

#### Explanation

Only recently I discovered that `bash -c 'set -e; ! true; echo ok'` prints `ok` (`bash --version`: `4.4.12(1)-release`). (See https://stackoverflow.com/a/39582012 for what's the problem. Instead of `true`, things like `grep something_unexpected some_file` can be used in real life. Well, actually such things had *been* used in some of my scripts before I discovered the gotcha.)

So `shellcheck` could report cases like `! thing` (out of `if` and so on) after `set -e` (or probably regardless of `set -e` or another `set`). Like: "`!` is probably unintentional". (Although it can't be said that `!` has not effect here: e.g., `$?` is affected. But still I think that the `set -e; ! true` gotcha is more common than such intentional cases.)

**Comment by koalaman** [comment]
I didn't know this! Good suggestion, thanks.

The final check triggered pretty rarely at ~0.8 files per 1000, but they all seemed like true positives and the non-directive workaround of doing `{ ! cmd; }` is simple and straight forward.
